src/gui/programupdater.cpp

```diff
@@ -190,28 +211,44 @@ void ProgramUpdater::fallbackDownloadFinished(const Net::DownloadResult &result)
 
     if (const QJsonValue verJSON = json[platformKey][u"version"_s]; verJSON.isString())
     {
-        const ProgramUpdater::Version tmpVer {verJSON.toString()};
+        const Version tmpVer {verJSON.toString()};
         if (isVersionMoreRecent(tmpVer))
-            m_fallbackRemoteVersion = tmpVer;
+            version = tmpVer;
     }
 
     handleFinishedRequest();
 }
 
 bool ProgramUpdater::updateProgram() const
 {
-    return QDesktopServices::openUrl(shouldUseFallback() ? u"https://www.qbittorrent.org/download"_s : m_updateURL);
+    switch (selectRemote())
+    {
+    case RemoteSources::Fosshub:
+        return QDesktopServices::openUrl(m_updateURL);
+    case RemoteSources::QbtMain:
+        return QDesktopServices::openUrl(u"https://www.qbittorrent.org/download"_s);
+    case RemoteSources::QbtBackup:
+        return QDesktopServices::openUrl(u"https://qbittorrent.github.io/qBittorrent-website/download"_s);
+    }
+    Q_UNREACHABLE();
 }
 
 void ProgramUpdater::handleFinishedRequest()
 {
-    if (m_hasCompletedOneReq)
+    ++m_fetchCompletedCount;
+    if (m_fetchCompletedCount == 3)
         emit updateCheckFinished();
-    else
-        m_hasCompletedOneReq = true;
+    Q_ASSERT((m_fetchCompletedCount >= 0) && (m_fetchCompletedCount <= 3));
```

Comment:
IMO, more flexible way would be to count pending requests instead. Since you're making all the requests in one place at the same time, you could set the count of pending requests there and then decrease it in this function and compare it to zero.
(Optional)

Reply:
OK. All comments addressed.